deploy/terraform/main.tf

```diff
@@ -0,0 +1,166 @@
+# that way we all pull the same state file when someone makes changes to TF.
+
+terraform {
+  required_providers {
+    aws = {
+      source = "hashicorp/aws"
+      version = "~> 4.53.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = var.AWS_REGION
+  profile = "opensource"
+}
+##################################################################################
+# DATA
+##################################################################################
+
+data "aws_vpc" "opensource_vpc" {
+  id = var.vpc_id
+}
+
+data "aws_subnet" "subnet_1" {
+  id = var.subnet_id
+}
+
+data "aws_ssm_parameter" "ami" {
+  name = "/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2"
+}
+
+##################################################################################
+# RESOURCES
+##################################################################################
+
+# Container service #
+#resource "aws_ecs_service" "osmt_container" {
+#  name = "osmt_app"
+#}
+
+############ DATABASES ##########
+resource "aws_db_instance" "osmt_db" {
+  allocated_storage    = 20
+  db_name              = "osmt_db"
+  engine               = "mysql"
+  engine_version       = "8.0"
+  instance_class       = "db.t3.micro"
+  username             = "osmt_user"
+  password             = var.db_password
+  port                 = 3306
+  identifier           = "osmt-dev-rds"
+  auto_minor_version_upgrade = true
+  skip_final_snapshot  = true
+  vpc_security_group_ids = [aws_security_group.rds_sg]
+}
+
+resource "aws_elasticache_cluster" "redis_db" {
+  cluster_id           = "redis-task-queue"
+  engine               = "redis"
+  node_type            = "cache.r6gd.xlarge"
+  num_cache_nodes      = 1
+  parameter_group_name = "default.redis6.0.6"
+  engine_version       = "6.0"
+  port                 = 6379
+}
+
+######### SECURITY GROUPS #########
+resource "aws_security_group" "rds_sg" {
+  name = "rds-sg"
+  description = "RDS security group"
+  vpc_id = data.aws_vpc.opensource_vpc.id
+
+  ingress {
+    from_port = 3306
+    protocol  = "tcp"
+    to_port   = 3306
+  }
+}
+
+resource "aws_security_group" "elasticsearch_sg" {
+  name = "elasticsearch-sg"
+  description = "ElasticSearch instance security group"
+  vpc_id = data.aws_vpc.opensource_vpc.id
+
+  ingress {
+    from_port = 9200
+    protocol  = "tcp"
+    to_port   = 9200
+    cidr_blocks = [data.aws_vpc.opensource_vpc.cidr_block]
+  }
+
+  egress {
+    from_port = 0
+    protocol  = -1
+    to_port   = 0
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+########## EC2 INSTANCES ##########
+resource "aws_instance" "elasticsearch" {
+  ami                     = nonsensitive(data.aws_ssm_parameter.ami.value)
+  count                   = 2
+  instance_type           = "t2.micro"
+  subnet_id               = data.aws_subnet.subnet_1.id
+  vpc_security_group_ids  = [aws_security_group.elasticsearch_sg.id]
+  root_block_device {
+    volume_size = 40
+    volume_type = "gp3"
+  }
+  user_data = <<EOF
```

Comment:
There are some env vars in this shell script that exist as if through magic. Where do these come from?

- $elasticsearch_version
- $REGION
- $ip_list
- $IPV4